### PR TITLE
feat: Implements helper methods for vectors/values

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Implements helper methods for vectors/values, including
- `is_xxx()` in ConcreteDataType
- `arrow_array_get()`
- `LogicalTypeId::data_type()`
- `TryFrom<ScalarValue> for Value`
- methods in `Helper`

Also passes all test cases

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #555

### CODE REVIEW

File: src/datatypes2/src/vectors.rs [hunk 1]
```diff
@@ -42,10 +42,11 @@ mod validity;
 
 pub use binary::*;
 pub use boolean::*;
+pub use constant::*;
 pub use date::*;
 pub use datetime::*;
 pub use helper::Helper;
-pub use list::*;
+pub use list::{ListIter, ListVector, ListVectorBuilder}; // Avoid re-exporting the test utils.
```
Comment (waynexia) [inline src/datatypes2/src/vectors.rs:49]: Also explicitly expand other `*`? It's not recommended in our style guide.

File: src/datatypes2/src/type_id.rs [hunk 2]
```diff
@@ -58,30 +58,36 @@ impl LogicalTypeId {
...
+            LogicalTypeId::TimestampMillisecond => {
+                ConcreteDataType::timestamp_millisecond_datatype()
+            }
+            LogicalTypeId::TimestampMicrosecond => {
+                ConcreteDataType::timestamp_microsecond_datatype()
+            }
+            LogicalTypeId::TimestampNanosecond => ConcreteDataType::timestamp_nanosecond_datatype(),
+            LogicalTypeId::List => {
+                ConcreteDataType::list_datatype(ConcreteDataType::null_datatype())
+            }
```
Comment (waynexia) [inline src/datatypes2/src/type_id.rs:90]: Are those braces needless?

File: src/datatypes2/src/data_type.rs [hunk 3]
```diff
@@ -61,66 +61,67 @@ pub enum ConcreteDataType {
...
+            self,
+            ConcreteDataType::Float64(_) | ConcreteDataType::Float32(_)
+        )
+    }
+
+    pub fn is_boolean(&self) -> bool {
+        matches!(self, ConcreteDataType::Boolean(_))
+    }
+
+    pub fn stringifiable(&self) -> bool {
```
Comment (waynexia) [inline src/datatypes2/src/data_type.rs:78]:
```suggestion
    pub fn is_stringifiable(&self) -> bool {
```

To keep consistent with other predicates.

File: src/datatypes2/src/type_id.rs (hunk 2 above)
Comment (evenyag) [reply src/datatypes2/src/type_id.rs:90]: They are added by rustfmt

**Review by waynexia — APPROVED** [review]

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**